src/core/vectortile/qgsmapboxglstyleconverter.cpp

```diff
@@ -2902,14 +2967,22 @@ QString QgsMapBoxGlStyleConverter::retrieveSpriteAsBase64( const QVariant &value
 
 QString QgsMapBoxGlStyleConverter::parseValue( const QVariant &value, QgsMapBoxGlStyleConversionContext &context )
 {
+  QColor c;
+  QString debug;
   switch ( value.type() )
   {
     case QVariant::List:
     case QVariant::StringList:
       return parseExpression( value.toList(), context );
 
+    case QVariant::Bool:
```

Comment:
I think bool should have it's own logic here -- a bool value will never be convertible to a color value.

Reply:
Now that the string to color conversion is out of parseValue, conversion from bool to string is handled in QgsExpression::quotedValue